dashboard/test/lib/school_info_interstitial_helper_test.rb

```diff
@@ -1,205 +1,198 @@
 require 'test_helper'
+require 'timecop'
 
 # Tests following spec:
 # We should pop up this interstitial again every 7 days unless one of the following is true:
 # - The teacher already filled out all the fields (except location)
 # - They chose “Homeschool,” “After School”, "Organization" or “Other” for school type
 # - Teacher chose a country that’s not “United States”
 class SchoolInfoInterstitialHelperTest < ActiveSupport::TestCase
-  test 'complete if all school info is provided' do
-    school_info = SchoolInfo.new
-    school_info.country = 'United States'
-    school_info.school_type = SchoolInfo::SCHOOL_TYPE_PUBLIC
-    school_info.school_name = 'Primary School'
-    school_info.full_address = '123 Sesame Street'
-
-    assert_nil school_info.school_id
-    refute_nil school_info.school_name
-    refute_nil school_info.full_address
-    assert SchoolInfoInterstitialHelper.complete? school_info
+  setup do
+    Timecop.freeze
   end
 
-  test 'complete if all school info but location is provided' do
-    school_info = SchoolInfo.new
-    school_info.country = 'United States'
-    school_info.school_type = SchoolInfo::SCHOOL_TYPE_PUBLIC
-    school_info.school_name = 'Primary School'
-
-    assert_nil school_info.school_id
-    refute_nil school_info.school_name
-    assert_nil school_info.full_address
-
-    assert SchoolInfoInterstitialHelper.complete? school_info
-  end
-
-  test 'complete if school is found by NCES id' do
-    school_info = SchoolInfo.new
-    school_info.school_id = 1
-
-    refute_nil school_info.school_id
-    assert SchoolInfoInterstitialHelper.complete? school_info
+  teardown do
+    Timecop.return
   end
 
-  test 'complete if school type is homeschool/after school/organization/other' do
-    school_info = SchoolInfo.new
-    school_info.country = 'United States'
-
-    school_info.school_type = SchoolInfo::SCHOOL_TYPE_HOMESCHOOL
-    assert SchoolInfoInterstitialHelper.complete? school_info
-
-    school_info.school_type = SchoolInfo::SCHOOL_TYPE_AFTER_SCHOOL
-    assert SchoolInfoInterstitialHelper.complete? school_info
+  test 'does not show a dialog if user is not a teacher' do
+    user = create :user
```

Comment:
nit: would it be better to explicitly create a student here?

Reply:
I kind of like it as-is. We assert the salient point below (that the created user is not a teacher) and this lets the test care less about the full set of possible user types.